guidance/src/guidance/guidance_worker.cpp

```diff
@@ -157,6 +157,17 @@ namespace guidance
         }
     }
 
+    void GuidanceWorker::create_guidance_state_machine()
+    {
+        if(vehicle_type == "cadilac") {
+            gsm = guidance_state_machine_factory.createCadilacInstance();
+        }
+        else if (vehicle_type == "lexus") {
+            gsm = guidance_state_machine_factory.createLexusInstance();
+        }
```

Comment:
Not a required change, but I think generally with factory pattern the standard practice is to put this logic inside the factory such that you would call 
guidance_state_machine_factory.createInstance("lexus").